2.6/root/usr/share/container-scripts/mongodb/common.sh

```diff
@@ -67,12 +66,11 @@ function wait_for_mongo_up() {
 
 # wait_for_mongo_down waits until the mongo server is down
 function wait_for_mongo_down() {
+  local i
   for i in $(seq $MAX_ATTEMPTS); do
     echo "=> Waiting for MongoDB service shutdown ..."
-    set +e
-    mongo admin --eval "help" &>/dev/null
-    status=$?
-    set -e
+    local status=0
+    mongo admin --eval 'help' &>/dev/null || status=$?
```

Comment:
This works, but seems unnecessarily complex to me.
If the `status=$?` part is ever executed, we know that the if condition in the next line will be always true.

We don't need status if we execute `mongo` as the if condition. Suggestion:

```bash
# wait_for_mongo_down waits until the mongo server is down
function wait_for_mongo_down() {
  local i
  for i in $(seq $MAX_ATTEMPTS); do
    echo "=> Waiting for MongoDB service shutdown ..."
    if ! mongo admin --eval "help" &>/dev/null; then
      echo "=> MongoDB service has stopped"
      return 0
    fi
    sleep $SLEEP_TIME
  done
  echo "=> Giving up: Failed to stop MongoDB service"
  exit 1
}
```

I don't know what's the canonical way to format the code when using "!" (not) in front of a command...

In pseudocode, maybe I'd even find it easier to reason about something flatter like:

```
print "Waiting ..."
tries = 5
while mongo_is_up() && tries > 0:
   sleep 1
   tries--
if mongo_is_up():
   print "Giving up..."
   return 1
else
   print "Mongo has stopped"
   return 0
```

But again this code is also subject of another PR...